tests/python/pants_test/base/test_hash_utils.py

```diff
@@ -82,3 +86,83 @@ def check_bad_spec(spec):
+
+  def test_normal_object_encoding(self):
+    self.assertEqual(self._coercing_json_encode({}), '{}')
+    self.assertEqual(self._coercing_json_encode(()), '[]')
+    self.assertEqual(self._coercing_json_encode([]), '[]')
+    self.assertEqual(self._coercing_json_encode(set([])), '[]')
+    self.assertEqual(self._coercing_json_encode([{}]), '[{}]')
+    self.assertEqual(self._coercing_json_encode([('a', 3)]), '[["a", 3]]')
+    self.assertEqual(self._coercing_json_encode({'a': 3}), '{"a": 3}')
+    self.assertEqual(self._coercing_json_encode([{'a': 3}]), '[{"a": 3}]')
+    self.assertEqual(self._coercing_json_encode(set([1])), '[1]')
+
+  def test_rejects_ordered_dict(self):
+    with self.assertRaisesRegexp(TypeError, r'CoercingEncoder does not support OrderedDict inputs'):
+      self._coercing_json_encode(OrderedDict([('a', 3)]))
+
+  def test_non_string_dict_key_coercion(self):
+    self.assertEqual(self._coercing_json_encode({('a', 'b'): 'asdf'}),
+                     '{"[\\"a\\", \\"b\\"]": "asdf"}')
+
+  def test_string_like_dict_key_coercion(self):
+    self.assertEqual(self._coercing_json_encode({'a': 3}), '{"a": 3}')
+    self.assertEqual(self._coercing_json_encode({b'a': 3}), '{"a": 3}')
+
+  def test_nested_dict_key_coercion(self):
+    self.assertEqual(self._coercing_json_encode({(1,): {(2,): 3}}),
+                     '{"[1]": {"[2]": 3}}')
+
+  def test_collection_ordering(self):
+    self.assertEqual(self._coercing_json_encode(set([2, 1, 3])), '[1, 2, 3]')
+    self.assertEqual(self._coercing_json_encode({'b': 4, 'a': 3}), '{"a": 3, "b": 4}')
+    self.assertEqual(self._coercing_json_encode([('b', 4), ('a', 3)]), '[["b", 4], ["a", 3]]')
+    self.assertEqual(self._coercing_json_encode([{'b': 4, 'a': 3}]),
+                     '[{"b": 4, "a": 3}]' if PY3 else '[{"a": 3, "b": 4}]')
+
+
+class JsonHashingTest(unittest.TestCase):
+
+  def test_known_checksums(self):
+    """Check a laundry list of supported inputs to stable_json_sha1().
+
+    This checks both that the method can successfully handle the type of input object, but also that
+    the hash of specific objects remains stable.
+    """
+    self.assertEqual(stable_json_sha1({}), 'bf21a9e8fbc5a3846fb05b4fa0859e0917b2202f')
+    self.assertEqual(stable_json_sha1(()), '97d170e1550eee4afc0af065b78cda302a97674c')
+    self.assertEqual(stable_json_sha1([]), '97d170e1550eee4afc0af065b78cda302a97674c')
+    self.assertEqual(stable_json_sha1(set([])), '97d170e1550eee4afc0af065b78cda302a97674c')
+    self.assertEqual(stable_json_sha1([{}]), '4e9950a1f2305f56d358cad23f28203fb3aacbef')
+    self.assertEqual(stable_json_sha1([('a', 3)]), 'd6abed2e53c1595fb3075ecbe020365a47af1f6f')
+    self.assertEqual(stable_json_sha1({'a': 3}), '9e0e6d8a99c72daf40337183358cbef91bba7311')
+    self.assertEqual(stable_json_sha1([{'a': 3}]), '8f4e36849a0b8fbe9c4a822c80fbee047c65458a')
+    self.assertEqual(stable_json_sha1(set([1])), 'f629ae44b7b3dcfed444d363e626edf411ec69a8')
```

Comment:
Set literal

```suggestion
    self.assertEqual(stable_json_sha1({1}), 'f629ae44b7b3dcfed444d363e626edf411ec69a8')
```

Also you may be wondering why I'm not suggesting a literal for `set([])`? This is because `{}` is a dict, but `{foo}` is a set. So there we do have to use `set([])` (or `set()`).

Reply:
> This is because `{}` is a dict, but `{foo}` is a set.

I love python because it's so consistent(ly annoying)!